### PR TITLE
Sort plurals forms in CLDR order

### PR DESCRIPTION
**Pull request by dwaynebailey**
Instead of alphabetical, sort these in the expect CLDR order zero...other

Only sort when retreiving the values for formatter consumption, this should leave Twine files in their default order.

### CODE REVIEW

File: lib/twine/twine_file.rb [hunk 1]
```diff
@@ -58,7 +58,7 @@ def translation_for_lang(lang)
 
     def plural_translation_for_lang(lang)
       if @plural_translations.has_key? lang
-        @plural_translations[lang].dup
+        @plural_translations[lang].dup.sort_by { |key,_| TwineDefinition::PLURAL_KEYS.index(key) }.to_h
```
Comment (rtsisyk) [inline lib/twine/twine_file.rb:61]: Thanks for addressing issues related to plurals. Could you please provide some examples of how it affects generated files in the main repo?

Comment (dwaynebailey) [reply lib/twine/twine_file.rb:61]: Sure, was hoping to point the consumption PR in the main OM repo. But here are some snippets.

The ordering is moving from alphabetic order to CLDR order, most are correct and only UK, PL, BE and RU are impacted:

Alphabetic: few many one other two zero (current)
CLDR: zero one two few many other (this PR)

Belarusian Android

```diff
diff --git a/android/app/src/main/res/values-be/strings.xml b/android/app/src/main/res/values-be/strings.xml
index 543aa2bc28..02c2c51571 100644
--- a/android/app/src/main/res/values-be/strings.xml
+++ b/android/app/src/main/res/values-be/strings.xml
@@ -602,8 +602,8 @@
        <string name="bookmark_lists_hide_all">Схаваць усе</string>
        <string name="bookmark_lists_show_all">Паказаць усе</string>
        <plurals name="bookmarks_places">
-         <item quantity="few">%d закладкі</item>
          <item quantity="one">%d закладка</item>
+         <item quantity="few">%d закладкі</item>
          <item quantity="other">%d закладак</item>
        </plurals>
        <string name="bookmarks_create_new_group">Стварыць новы спіс</string>
@@ -622,8 +622,8 @@
        <string name="profile">Профіль OpenStreetMap</string>
        <string name="restore">Аднавіць</string>
        <plurals name="tracks">
-         <item quantity="few">%d сцежкі</item>
          <item quantity="one">%d сцежка</item>
+         <item quantity="few">%d сцежкі</item>
          <item quantity="other">%d сцежак</item>
        </plurals>
        <!-- Settings privacy group in settings screen -->
```

And Russian, Apple plurals

```diff
diff --git a/iphone/Maps/LocalizedStrings/ru.lproj/Localizable.stringsdict b/iphone/Maps/LocalizedStrings/ru.lproj/Localizable.stringsdict
index f932aa6f17..61d5263355 100644
--- a/iphone/Maps/LocalizedStrings/ru.lproj/Localizable.stringsdict
+++ b/iphone/Maps/LocalizedStrings/ru.lproj/Localizable.stringsdict
@@ -19,10 +19,10 @@
       <string>NSStringPluralRuleType</string>
       <key>NSStringFormatValueTypeKey</key>
       <string>d</string>
-      <key>few</key>
-      <string>%d метки</string>
       <key>one</key>
       <string>%d метка</string>
+      <key>few</key>
+      <string>%d метки</string>
       <key>other</key>
       <string>%d меток</string>
     </dict>
```

**Comment by dwaynebailey** [comment]
This is the full range of changes in the OM main repo PR https://github.com/organicmaps/organicmaps/pull/8959/files

**Review by rtsisyk — APPROVED** [review]
LGTM. Thanks!!

**Review by biodranik — APPROVED** [review]
Thanks! Is it for clarity only, or is it a pre-requisite for something else?

**Comment by dwaynebailey** [comment]
> Thanks! Is it for clarity only, or is it a pre-requisite for something else?

Both:
1. Its clearer to see what forms are present, by following the expected order
2. It reduces the diff when comparing the output of other tools, the noise hides any potential gotchas

**Pull request merged**